### ISSUE TITLE
Multiple file upload - Got 400

### ISSUE THREAD

**Issue by jy95** [cdimascio/express-openapi-validator#176]
Hello,

In Open API V3 specs, it is possible to specify how multiple file upload is handled ( https://swagger.io/docs/specification/describing-request-body/file-upload/ ). 
```yaml
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                filename:
                  type: array
                  items:
                    type: string
                    format: binary
```

After that, I documented the following endpoint ( that does a bulk insert of exercises with their zip files sources ) : 

```yaml
  /api/bulk_create_exercises:
    post:
      summary: "Creates multiple exercises into the system"
      operationId: createMultipleExercises
      security:
        - bearerAuth: []
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/MultipleExercisesForm"
          multipart/form-data:
            schema:
              type: object
              properties:
                exercisesData:
                  $ref: "#/components/schemas/MultipleExercisesForm"
                files:
                  type: array
                  description: "The sources of the exercises, each in zip format"
                  minItems: 1
                  items:
                    type: string
                    format: binary
                    description: "The source of an exercise, in zip format"
                filesMapping:
                  type: array
                  description: "Mapping between the given file (to find out which data belong to exercise)"
                  minItems: 1
                  items:
                    type: object
                    properties:
                      filename:
                        type: string
                        description: "The name of the given source file"
                        example: "file1.zip"
                      exercise:
                        type: integer
                        description: "The location of the related exercise in the exercisesData array"
                    required:
                      - filename
                      - exercise
              required:
                - exercisesData
                - files
                - filesMapping
      responses:
        '200':
          description: OK
        # Definition of all error statuses
        default:
          description: "Whatever error : 4XX - Client error (Bad Request, Unauthorized, etc.) , 5XX - Server error"
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/ErrorObject"
```

I expected this to work but I got a nice 400 error : 
![image](https://user-images.githubusercontent.com/9306961/70566149-cc4d9380-1b93-11ea-8ddf-08264664b16f.png)

When I look up inside the req before my error handler throws the 400 status code, I see that : 

![upload3](https://user-images.githubusercontent.com/9306961/70566313-1afb2d80-1b94-11ea-8138-f669aa3a966e.PNG)

And the files are correctly uploaded : 

![upload4](https://user-images.githubusercontent.com/9306961/70566333-277f8600-1b94-11ea-95e1-9795964b9fc8.PNG)

I checked in the tests and there is only single upload [tests ](https://github.com/cdimascio/express-openapi-validator/blob/master/test/multipart.spec.ts)

Thanks for your help ;)

**Comment by jy95** [comment]
If you need a supertest request example : 
```js

// an array of exercises
const exercises = [{
  "title": "An example",
  "description": "I love bugs XD",
  "tags": [{
    "autoGenerated": true,
    "category_id": "_PLATFORM_",
    "text": "INGINIOUS"
  }]
}]

// files linked to exercises
const files = [{
  "filename": "someZipFile.zip",
  "path": "/path/to/the/file",
  "exercise": 0
}]

// build the request now
let requestInstance = request
        .post("/api/bulk_create_exercises")
        .set('Authorization', 'bearer ' + JWT_TOKEN);

// Add all given files
for (const file of files) {
  requestInstance.attach("files", file.path)
}

// Add the mapping between exercises and files
files.forEach((file, index) => {
  const sub_field = "filesMapping[" + index + "]";
  requestInstance.field(sub_field + "[filename]", file.filename);
  requestInstance.field(sub_field + "[exercise]", file.exercise);
});

// Add the exercises metadata
exercises.forEach((exercise, index) => {

  // since tags are more complicated to deal with, I must handle them separately
  const exercise_tags = exercise.tags;
  delete exercise.tags;

  const sub_field = "exercisesData[" + index + "]";

  // for other properties of exercise, it is pretty easy to handle them
  Object.entries(exercise).forEach(([key, value]) => {
    requestInstance.field(sub_field + "[" + key + "]", value);
  });

  // for tags, we have to use this ugly way because of supertest
  const sub_tag_field = sub_field + "[tags]";

  exercise_tags.forEach((tag, index) => {
    const sub_tag_field_index = sub_tag_field + "[" + index + "]";
    Object.entries(tag).forEach(([key, value]) => {
      requestInstance.field(sub_tag_field_index + "[" + key + "]", value);
    });
  });
});
```

**Comment by cdimascio** [comment]
Thanks for all the detail. 
Currently, only 1 file is able to be uploaded. It's possible we'll add support for more cases. That being said, complex upload cases are out of arguably out of scope for the validator. 

As a result, I am planning to make it possible to opt-out of file upload so that folks with more complex use cases, can manage those cases themselves. In the meantime, you can use `ignorePaths` to skip validation on your multi-file upload endpoint, so that you can implement your use case without the validator getting in your way. 

That being said, if you do solve the multi-upload case using `multer`, please send me the code. Perhaps, we can use that as the basis for multi-file upload support. Also, if you're feeling inspired want to take a pass at implementing multi-file upload in the, feel free to submit a PR.

**Comment by jy95** [comment]
Is it just possible to ignore the `files` types in validation only ? 
The problem is that the validator tried to validate this field whereas it is `''` because of multer (it also occurs for a single upload endpoint ) 

![image](https://user-images.githubusercontent.com/9306961/70610943-b5935500-1c04-11ea-8fe5-821ed5260c20.png)

**Comment by jy95** [comment]
@cdimascio I think I found a nice way to allow multiple uploads (and keep the single upload to work also ) - the PR awaits you ^^

**Comment by cdimascio** [comment]
@jy95 thanks for the PR! your work is a nice improvement over what is current. the change is merged and available in `v3.2.2`. thanks again!

**Comment by kulkarnipradnyas** [comment]
from request body i am sending a property files with correct type:string format:binary but still my validator is failing with 500 and in error as well in body object i cant see files property.I am not able to upload file.what is the solution? using latest version of this package.
